### ISSUE TITLE
Number of pieces on user profile not updatingand image not appearing on map

### ISSUE THREAD

**Issue by lizheard1** [PlasticPatrol/PlasticPatrolWeb#128]
Katie - who has been testing on Android, has uploaded several images, of which I have approved all of them.

Her user profile still says: 
Num. uploads: 0 
Total Pieces: 0
0
And she cannot find her images on the map.
However she appears on the leader board with 82 pieces (ie the leaderboard has updated but not her profile and not on the map)

Email address associated with her account: katiemawdsley@me.com

**Comment by munroTom** [comment]
Haven't solved it just yet - but think it's an issue with browser caching

If I visit https://plastic-patrol-dev-722eb.web.app/#/@53.7745501,-2.7562463,17.26z I can't see them, but if I visit it incognito then her photos appear for me

**Comment by neilfulwiler** [comment]
@munroTom yeah we cache geojson pretty aggressively and AFAICT don't really ever re-load it in it's entirety. However, we *do* set up a subscription to it so that if something happens in real time we'll actually get the update, but that's pretty infrequent. I thought about adding a timestamp to the cached geojson object and trying to do this the right way (also I sort of thought that firebase would handle this sort of thing for us transparently), but I decided that the easiest thing here is just to issue a separate query for all of the user's photos.

@lizheard1 that means that the caching issue in general isn't solved (if we removed the cache things would be much worse, so it does help us), but users shouldn't be affected by it when trying to view their own data, which we bypass cache for. Does that sound good to you? (for example, this means that if they try to view their friends profile/images, they'll still be stale)

**Comment by lizheard1** [comment]
@neilfulwiler does this fix mean that:
a) a user can see the number of pieces on their personal profile, and this updates once an image is approved?
b) a user can navigate the map and see all photos uploaded (if they zoom into another area) - we want people to be able to see the scale of images out there - the more images they can see, they feel they are having more impact. Also if litter is logged in new countries, we want people to be able to see that visually on the map and feel the global spread

If it works for the above, then great!

Also, how regularly does the map update and profile stats update?

**Comment by lizheard1** [comment]
Reopening this as I've had a few messages from people saying their images aren't appearing
